File: archive/z/zig/fizz-buzz.zig
const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    var i: usize = 1;
    while (i <= 100) : (i += 1) {
        if (i % 15 == 0) {
            try stdout.writeAll("FizzBuzz\n");
        } else if (i % 3 == 0) {
            try stdout.writeAll("Fizz\n");
        } else if (i % 5 == 0) {
            try stdout.writeAll("Buzz\n");
        } else {
            try stdout.print("{d}\n", .{i});
        }
    }
}

File: archive/c/c-plus-plus/fizz-buzz.cpp
#include <iostream>

int main() {
    for (int i = 1; i <= 100; ++i) {
        const bool fizz = (i % 3 == 0);
        const bool buzz = (i % 5 == 0);

        if (fizz && buzz) {
            std::cout << "FizzBuzz\n";
        } else if (fizz) {
            std::cout << "Fizz\n";
        } else if (buzz) {
            std::cout << "Buzz\n";
        } else {
            std::cout << i << '\n';
        }
    }
    
    return 0;
}
